### PR TITLE
Improve Application Settings `Contacts` field & Fix Home page cards responsive issues

### PR DESCRIPTION
**Pull request by brionmario**
### Purpose

This pull request refactors the Contacts section in the application customization settings to provide a more user-friendly and robust experience for entering contact email addresses. Instead of a multiline text field, users now add emails as chips using an autocomplete input, with validation and clearer feedback. The update also introduces a reusable `Kbd` component for rendering keyboard key labels and updates related tests and translations accordingly.

**Contacts Section Redesign and Validation:**

- Replaced the multiline text field for contacts with a creatable `Autocomplete` input that allows users to add email addresses as chips; emails are validated on entry, and invalid emails are rejected with an inline error message. The UI now uses a placeholder and hint text to guide users, and the list of emails is synced as an array. (`frontend/apps/thunder-develop/src/features/applications/components/edit-application/customization-settings/ContactsSection.tsx`, [[1]](diffhunk://#diff-7f35622b8530757cfdbb9076d7b73ce81c71fe0677c8ab3c85b306356619071bL19-R24) [[2]](diffhunk://#diff-7f35622b8530757cfdbb9076d7b73ce81c71fe0677c8ab3c85b306356619071bL49-R105)
- Updated i18n translations to reflect the new input method, including a dynamic hint that uses the new `Kbd` component to display the "Enter" key, and added an error message for invalid emails. (`frontend/packages/thunder-i18n/src/locales/en-US.ts`, [frontend/packages/thunder-i18n/src/locales/en-US.tsL1165-R1167](diffhunk://#diff-964951f5cd4b5cc3d586157f068d11adf59213bef8e82dd2444f20985ae03055L1165-R1167))

**New Reusable Component:**

- Added a new `Kbd` component for rendering keyboard key labels in a stylized manner, improving accessibility and clarity in helper texts and documentation. (`frontend/apps/thunder-develop/src/components/Kbd.tsx`, [frontend/apps/thunder-develop/src/components/Kbd.tsxR1-R54](diffhunk://#diff-00ebd25354e57c4da3f6ace3eb4129addb1671a408d276ddfcc7e11d0d2a3c3eR1-R54))

**Testing Enhancements:**

- Refactored and expanded the Contacts section tests to cover the new autocomplete and chip UI, email validation logic, error handling, and edge cases, ensuring robust coverage of the new behavior. (`frontend/apps/thunder-develop/src/features/applications/components/edit-application/customization-settings/__tests__/ContactsSection.test.tsx`, [[1]](diffhunk://#diff-e52508fa89a9becc9b9428a94ba7aff00f756cb2ff312ce64c352c0cb1fab006L20-R20) [[2]](diffhunk://#diff-e52508fa89a9becc9b9428a94ba7aff00f756cb2ff312ce64c352c0cb1fab006R29) [[3]](diffhunk://#diff-e52508fa89a9becc9b9428a94ba7aff00f756cb2ff312ce64c352c0cb1fab006L47-L199)
- Added tests for the new `Kbd` component to verify rendering and styling. (`frontend/apps/thunder-develop/src/components/__tests__/Kbd.test.tsx`, [frontend/apps/thunder-develop/src/components/__tests__/Kbd.test.tsxR1-R83](diffhunk://#diff-8d0fa1dcf57cd90e24595916af45dbc615aa7cc1e3b01ac7fd140a3980d8c867R1-R83))

**Minor UI Improvement:**

- Ensured that cards in the Next Steps section always fill their grid cell height for improved layout consistency. (`frontend/apps/thunder-develop/src/features/home/components/NextStepsSection.tsx`, [frontend/apps/thunder-develop/src/features/home/components/NextStepsSection.tsxL59-R74](diffhunk://#diff-76480020d4aad92561b3123ea4b0872a98f82e2d82b4bd4a3f61b0a140ce74c8L59-R74))

### Related Issues
- https://github.com/asgardeo/thunder/issues/1795

### Related PRs
- N/A

### Checklist
- [ ] Followed the contribution guidelines.
- [ ] Manual test round performed and verified.
- [ ] Documentation provided. (Add links if there are any)
    - [ ] Ran Vale and fixed all errors and warnings
- [ ] Tests provided. (Add links if there are any)
    - [ ] Unit Tests
    - [ ] Integration Tests
- [ ] Breaking changes. (Fill if applicable)
    - [ ] Breaking changes section filled.
    - [ ] `breaking change` label added.

### Security checks
- [ ] Followed secure coding standards in [WSO2 Secure Coding Guidelines](https://security.docs.wso2.com/en/latest/security-guidelines/secure-engineering-guidelines/secure-coding-guidlines/introduction/)
- [ ] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets.

## Summary by CodeRabbit

* **New Features**
  * Contacts input redesigned with autocomplete chips, inline email validation, and improved helper text (includes keyboard hint).
  * Keyboard shortcut (Kbd) indicators added for clearer guidance.

* **Bug Fixes**
  * Improved inline error handling and clearing behavior for invalid email entries.

* **Translations**
  * Updated contact input hints and added new error messages for email validation and redirect URI validation.

### CODE REVIEW

File: tests/e2e/package.json [hunk 1]
```diff
@@ -50,5 +50,8 @@
   },
   "dependencies": {
     "dotenv": "17.2.3"
+  },
+  "overrides": {
+    "flatted": ">=3.4.1"
```
Comment (brionmario) [inline tests/e2e/package.json:55]:
```
❯ npm ls flatted
e2e-tests-thunder@1.0.0 /Users/brion/Desktop/thunder⚡/thunder-design-api copy/tests/e2e
└─┬ eslint@9.0.0
  └─┬ file-entry-cache@8.0.0
    └─┬ flat-cache@4.0.1
      └── flatted@3.3.3
```

`lat-cache` &  `file-entry-cache` hasn't updated yet.

Comment (brionmario) [reply tests/e2e/package.json:55]: https://github.com/advisories/GHSA-25h7-pfq9-p65f

Comment (brionmario) [reply tests/e2e/package.json:55]: Fix for`flat-cache` & `file-entry-cache`.
- https://github.com/jaredwray/cacheable/pull/1594
